services/staging/rs_server_staging/main.py

```
@@ -284,10 +286,20 @@ async def execute_process(req: Request, resource: str, data: ProcessMetadataMode
 @router.get("/jobs/{job_id}")
 async def get_job_status_endpoint(job_id: str = Path(..., title="The ID of the job")):
     """Used to get status of processing job."""
-    job = app.extra["process_manager"].get_job(job_id)
-    if job:
-        return job
-    raise HTTPException(status_code=HTTP_404_NOT_FOUND, detail=f"Job with ID {job_id} not found")
+    try:
+        return app.extra["process_manager"].get_job(job_id)
+    except JobNotFoundError:
+        # Handle case when job_id is not found
+        return JSONResponse(
+            status_code=HTTP_404_NOT_FOUND,
+            content={"title": "No Such Job", "detail": f"Job with ID {job_id} not found"},
+        )
+    except Exception as e:  # pylint: disable=broad-exception-caught
```

Do you really need these lines 297~ 302 ? You could remove them and just let the exception propagate.